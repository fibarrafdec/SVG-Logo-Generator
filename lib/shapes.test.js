// shapes.test.js

const inquirer = require("inquirer");
const { Circle, Triangle, Square } = require("./shapes");

// Mock the inquirer prompt function to provide user input for testing
jest.mock("inquirer");

describe("Shape Classes", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should create a Circle with user input", async () => {
    // Mock user input for Circle
    inquirer.prompt.mockResolvedValueOnce({
      shapeType: "Circle",
      shapeColor: "blue",
      text: "C",
      textColor: "white",
    });

    const shape = await createShape();
    expect(shape instanceof Circle).toBeTruthy();
    expect(shape.color).toBe("blue");
    expect(shape.radius).toBe(170);
    expect(shape.render("C", "white")).toMatchSnapshot();
  });

  it("should create a Triangle with user input", async () => {
    // Mock user input for Triangle
    inquirer.prompt.mockResolvedValueOnce({
      shapeType: "Triangle",
      shapeColor: "green",
      text: "T",
      textColor: "red",
    });

    const shape = await createShape();
    expect(shape instanceof Triangle).toBeTruthy();
    expect(shape.color).toBe("green");
    expect(shape.sideLength).toBe(400);
    expect(shape.render("T", "red")).toMatchSnapshot();
  });

  it("should create a Square with user input", async () => {
    // Mock user input for Square
    inquirer.prompt.mockResolvedValueOnce({
      shapeType: "Square",
      shapeColor: "yellow",
      text: "S",
      textColor: "black",
    });

    const shape = await createShape();
    expect(shape instanceof Square).toBeTruthy();
    expect(shape.color).toBe("yellow");
    expect(shape.sideLength).toBe(350);
    expect(shape.render("S", "black")).toMatchSnapshot();
  });
});

// Function to create a shape based on user input
async function createShape() {
  const userInput = await inquirer.prompt([
    {
      type: "list",
      name: "shapeType",
      message: "Choose a shape:",
      choices: ["Circle", "Triangle", "Square"],
    },
    {
      type: "input",
      name: "shapeColor",
      message: "Enter shape color (color keyword or hexadecimal number):",
    },
    {
      type: "input",
      name: "text",
      message: "Enter up to three characters (text):",
    },
    {
      type: "input",
      name: "textColor",
      message: "Enter text color (color keyword or hexadecimal number):",
    },
  ]);

  let shape;

  switch (userInput.shapeType) {
    case "Circle":
      shape = new Circle();
      break;
    case "Triangle":
      shape = new Triangle();
      break;
    case "Square":
      shape = new Square();
      break;
    default:
      throw new Error("Invalid shape type.");
  }

  shape.setColor(userInput.shapeColor);

  return shape;
}
