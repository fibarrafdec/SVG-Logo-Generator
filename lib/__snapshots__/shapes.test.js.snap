// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape Classes should create a Circle with user input 1`] = `
"
      <svg width="340" height="340" xmlns="http://www.w3.org/2000/svg">
        <circle cx="170" cy="170" r="170" fill="blue" />
        <text x="170" y="170" dominant-baseline="middle" text-anchor="middle" fill="white" font-size="80" font-weight="bold">C</text>
      </svg>
    "
`;

exports[`Shape Classes should create a Square with user input 1`] = `
"
      <svg width="350" height="350" xmlns="http://www.w3.org/2000/svg">
        <rect width="350" height="350" fill="yellow" />
        <text x="175" y="175" dominant-baseline="middle" text-anchor="middle" fill="black" font-size="80" font-weight="bold">S</text>
      </svg>
    "
`;

exports[`Shape Classes should create a Triangle with user input 1`] = `
"
      <svg width="400" height="346.41016151377545" xmlns="http://www.w3.org/2000/svg">
        <polygon points="0,346.41016151377545 400,346.41016151377545 200,0" fill="green" />
        <text x="200" y="173.20508075688772" dominant-baseline="middle" text-anchor="middle" fill="red" font-size="70" font-weight="bold">T</text>
      </svg>
    "
`;
